### DISCUSSION TITLE
OOP2 not receiving values on new phone (using Libre 2 + Miaomiao3)

### DISCUSSION THREAD

**Discussion by dagord** [NightscoutFoundation/xDrip#2634] [Q&A]
Hello,
my girlfriend is using Libre 2 + Miaomiao3 + OOP2 + xDrip.
She recently switched phone (Xiaomi Redmi Note 7) and I can't replicate the right settings; unfortunately the old phone is completely broken.
Basically, I installed OOP2 following the documentation here: https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip; anyway, OOP2 don't receive any reading; the Miaomiao is currently paired with xDrip and I suspect this is the problem.
The current sensor was started with the standalone reader, if this makes difference.
Both old and new phones don't support NFC.
Any suggestion is appreciated!

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
@dagord if you use MM you don't really care about NFC and which app started the sensor. Please confirm you use an European Libre 2 sensor though, and your miaomiao is running the latest firmware. You should have the latest version of OOP2 and OOP1 should be disabled https://xdrip.readthedocs.io/en/latest/use/OOP/#oop2

**Reply by dagord** [reply]
Looking into the logs it seems that the sensor wasn't recognized... so eventually disabled the OOP algorithm in xDrip settings and it started reading values!
So, even if it's a Libre2 sensor, it works as Libre1!
What's going on? I checked on the package and this sensor came from USA... I know that US and EU sensors are different.

![IMG_20230118_185626](https://user-images.githubusercontent.com/21258057/213292488-211c4482-379f-4882-9836-da6fbf86f9a8.jpg)

**Reply by psonnera** [reply]
Made in the USA but it's the european version. If you use a MM you can read it as a Libre 1.

**Reply by dagord** [reply]
Yes, in fact I had to set up xDrip for Libre1.
So what's the difference with "regular" Libre2 sensors? Sorry, I'm a bit confused

**Reply by psonnera** [reply]
This is a regular Libre 2 sensor. Other sensors can't be read by xDrip+ and OOPs.

**Reply by dagord** [reply]
Sorry, I don't understand what you are meaning.
I had to disable OOP2 to get this specific sensor working.

**Reply by psonnera** [reply]
@dagord Libre 2 don't work without OOP, disable OOP2 how? Then you have OOP1? Are you sure you didn't disable OOP in xDrip+ but OOP2 is still running?

**Reply by dagord** [reply]
Yes, OOP2 is running but I disabled "Out of process algorithm" option in Xdrip+ settings; if I enable it I stop get readings.

**Reply by dagord** [reply]
I also completely uninstalled OOP2 and the readings are working as usual

**Reply by psonnera** [reply]
@dagord so basically you mean you have no OOP installed on your phone? Can you share the events log showing a reading?
